storage/innobase/buf/buf0buf.cc

```diff
@@ -3766,14 +3766,21 @@ dberr_t buf_page_t::read_complete(const fil_node_t &node) noexcept
   if (err == DB_PAGE_CORRUPTED || err == DB_DECRYPTION_FAILED)
   {
 release_page:
-    if (node.space->full_crc32() && node.space->crypt_data &&
-        recv_recovery_is_on() &&
-        recv_sys.dblwr.find_encrypted_page(node, id().page_no(),
-                                           const_cast<byte*>(read_frame)))
+    if (node.space->full_crc32() && recv_recovery_is_on())
     {
-      /* Recover from doublewrite buffer */
-      err= DB_SUCCESS;
-      goto success_page;
+      if (node.space->crypt_data &&
+          recv_sys.dblwr.find_encrypted_page(node, id().page_no(),
+                                             const_cast<byte*>(read_frame)))
+       {
+dblwr_success:
+         /* Recover from doublewrite buffer */
+         err= DB_SUCCESS;
+         goto success_page;
+       }
+       else if (node.space->is_compressed() &&
+                recv_sys.dblwr.find_page_compressed(node, id().page_no(),
+                                                    const_cast<byte*>(read_frame)))
+         goto dblwr_success;
```

Comment:
The lines starting with `{` are indented too deep by 1 space.

If this is the only call site to the two functions, could we rename the member function to `find_deferred_page` and let it handle both these cases? It does not seem to be necessary to evaluate further conditions on `node.space` here; we could let the callee do that. After all, this code should be executed very rarely (`ATTRIBUTE_COLD`); we would want to minimize the overhead for normal page reads related to something that is only rarely needed during crash recovery.